frontend/app/routes/protected/person-case/secondary-doc.tsx

```diff
@@ -0,0 +1,179 @@
+
+import { faExclamationCircle, faXmark } from '@fortawesome/free-solid-svg-icons';
+import type { SessionData } from 'express-session';
+import { useTranslation } from 'react-i18next';
+import * as v from 'valibot';
+
+import type { Route, Info } from './+types/secondary-doc';
+
+import { requireAuth } from '~/.server/utils/auth-utils';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import { Button } from '~/components/button';
+import { DatePickerField } from '~/components/date-picker-field';
+import { FetcherErrorSummary } from '~/components/error-summary';
+import { InputFile } from '~/components/input-file';
+import { InputRadios } from '~/components/input-radios';
+import { PageTitle } from '~/components/page-title';
+import { Progress } from '~/components/progress';
+import { AppError } from '~/errors/app-error';
+import { ErrorCodes } from '~/errors/error-codes';
+import { getFixedT } from '~/i18n-config.server';
+import { handle as parentHandle } from '~/routes/protected/layout';
+import { getLanguage } from '~/utils/i18n-utils';
+
+type SecondaryDocumentsSessionData = NonNullable<SessionData['inPersonSINCase']['secondaryDocument']>;
+
+const VALID_DOCTYPE = ['passport', 'id-card', 'other-id'];
+
+export const handle = {
+  i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
+} as const satisfies RouteHandle;
+
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  return {
+    documentTitle: t('protected:secondary-identity-document.page-title'),
+    defaultFormValues: context.session.inPersonSINCase?.secondaryDocument,
+  };
+}
+
+export function meta({ data }: Route.MetaArgs) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: Route.ActionArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const lang = getLanguage(request);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  const formData = await request.formData();
+  const action = formData.get('action');
+  const maxImageSize = 1024 * 1024 * 15; //Max image size is 15 MB
+
+  switch (action) {
+    case 'back': {
+      throw i18nRedirect('routes/protected/person-case/primary-docs.tsx', request);
+    }
+    case 'next': {
+      const schema = v.object({
+        documentType: v.picklist(VALID_DOCTYPE, t('protected:secondary-identity-document.document-type.invalid')),
+        document: v.pipe(
+          v.file(t('protected:secondary-identity-document.upload-document.required')),
+          v.mimeType(
+            ['image/jpeg', 'image/png', 'image/heic'],
+            t('protected:secondary-identity-document.upload-document.invalid'),
+          ),
+          v.maxSize(maxImageSize),
+        ),
+        expiryDate: v.pipe(v.date(), v.minValue(new Date(), t('protected:secondary-identity-document.expiry-date.invalid'))),
```

Comment:
If we're going to invalidate documents by expiry dates, we should normalize `new Date()` to midnight at some predetermined timezone.

As-is, this code will behave differently depending on 1) the container timezone, 2) the time of day.

We need to ask business what timezone they want to use to determine when the day switches from today to tomorrow.

@sebastien-comeau

Reply:
I agree.